extensions/elasticsearch/elasticsearch-7/src/main/java/com/hazelcast/jet/elastic/ElasticSources.java

```diff
@@ -0,0 +1,135 @@
+ * Alternatively you can use {@link ElasticSourceBuilder}
+ *
+ * @since 4.1
+ */
+public final class ElasticSources {
+
+    private static final int DEFAULT_PORT = 9200;
+
+    private ElasticSources() {
+    }
+
+    /**
+     * Creates a source which queries local instance of Elasticsearch for all documents
+     * <p>
+     * Useful for quick prototyping. See other methods {@link #elastic(SupplierEx, SupplierEx, FunctionEx)}
+     * and {@link #builder()}
+     */
+    public static BatchSource<String> elastic() {
+        return elastic(() -> new RestHighLevelClient(
+                RestClient.builder(new HttpHost("localhost", DEFAULT_PORT))
+        ));
+    }
+
+    /**
+     * Creates a source which queries Elasticsearch using client obtained from {@link RestHighLevelClient} supplier.
+     * Queries all indexes for all documents.
+     * Uses {@link SearchHit#getSourceAsString()} as mapping function
+     */
+    public static BatchSource<String> elastic(@Nonnull SupplierEx<RestHighLevelClient> clientSupplier) {
+        return elastic(clientSupplier, SearchHit::getSourceAsString);
+    }
+
+    /**
+     * Creates a source which queries local instance of Elasticsearch for all documents
+     * Uses {@link SearchHit#getSourceAsString()} as mapping function
+     */
+    public static <T> BatchSource<T> elastic(@Nonnull FunctionEx<? super SearchHit, T> mapHitFn) {
+        return elastic(() -> new RestHighLevelClient(
+                        RestClient.builder(new HttpHost("localhost", DEFAULT_PORT))
+                ),
+                mapHitFn
+        );
+    }
+
+    /**
+     * Creates a source which queries Elasticsearch using client obtained from {@link RestHighLevelClient} supplier.
+     * Uses provided mapHitFn to map results.
+     * Queries all indexes for all documents.
+     *
+     * @param clientSupplier RestHighLevelClient supplier
+     * @param mapHitFn       supplier of a function mapping the result from SearchHit to a result type
+     * @param <T>            result type returned by the map function
+     */
+    public static <T> BatchSource<T> elastic(
+            @Nonnull SupplierEx<RestHighLevelClient> clientSupplier,
+            @Nonnull FunctionEx<? super SearchHit, T> mapHitFn) {
+        return elastic(clientSupplier, SearchRequest::new, mapHitFn);
+    }
+
+    /**
+     * Creates a source which queries Elasticsearch using client obtained from {@link RestHighLevelClient} supplier.
+     *
+     * @param clientSupplier        RestHighLevelClient supplier
+     * @param searchRequestSupplier supplier of a SearchRequest used to query for documents
+     * @param mapHitFn              supplier of a function mapping the result from SearchHit to a target type
+     * @param <T>                   result type returned by the map function
+     */
+    public static <T> BatchSource<T> elastic(
+            @Nonnull SupplierEx<RestHighLevelClient> clientSupplier,
```

Comment:
instead of client we should take the `RestClientBuilder`

Reply:
In our examples and tests we just pass the `RestClientBuilder` to `RestHighLevelClient`, but the user might have own subclass of RestHighLevelClient. 

I didn't investigate whether there is a use case where it would actually make sense to use a custom client. Will check and change it if not.

Reply:
Why would they subclass the client? I think the intention is to have the user write as little code as possible.

Reply:
Yeah, it probably doesn't make sense in our case.

Generally when you create a plugin to Elasticsearch and want to support it in the client you just subclass the client and add methods for your plugin. But the connector wouldn't use any of this so we are fine with the regular client.

Only thing might be authentication, but that's done either via the optionsFn or via the low level RestClient.

I will check if it can be replaced easily, there are some tests with the client mocked.

Reply:
this should use the builder, this hasn't been resolved.